### PR TITLE
Implemented source address support (IPL-43)

### PR DESCRIPTION
**Pull request by sbouchex**
Code proposal to specify a source address

### CODE REVIEW

File: src/com/trilead/ssh2/transport/TransportManager.java [hunk 1]
```diff
@@ -205,10 +206,11 @@ private InetAddress parseIPv4Address(String host) throws UnknownHostException
 		return InetAddress.getByAddress(host, addr);
 	}
 
-	public TransportManager(String host, int port) throws IOException
+	public TransportManager(String host, int port, String sourceAddress) throws IOException
```
Comment (mc1arke) [inline src/com/trilead/ssh2/transport/TransportManager.java]: This is public API so we need to keep the existing constructor signature and chain calls through to the new constructor as needed, e.g.

```java
public TransportManager(String host, int port) throws IOException
{
   this(host, port, null);
}

public TransportManager(String host, int port, String sourceAddress) throws IOException
{
   ...
}
```

File: src/com/trilead/ssh2/Connection.java [hunk 2]
```diff
@@ -137,6 +138,23 @@ public Connection(String hostname, int port)
...
+	 * @param hostname
+	 *            the host where we later want to connect to.
+	 * @param port
+	 *            port on the server, normally 22.
+	 */
+	public Connection(String hostname, int port, String sourceAddress)
+	{
+		this.hostname = hostname;
+		this.port = port;
+		this.sourceAddress = sourceAddress;
```
Comment (mc1arke) [inline src/com/trilead/ssh2/Connection.java]: You could have just chained the constructors here; it makes future changes simpler, but is no real issue.

**Comment by sbouchex** [comment]
@mc1arke I'll redo my PR to better suite the API and I'll rebase my PR as well.

**Comment by mc1arke** [comment]
@sbouchex the commits you've pushed today don't seem to have made any changes, are they supposed to make the modifications I'd requested?

File: src/com/trilead/ssh2/transport/TransportManager.java [hunk 3]
```diff
@@ -122,6 +122,7 @@ public void run()
 		}
 	}
 
+	String sourceAddress;
```
Comment (mc1arke) [inline src/com/trilead/ssh2/transport/TransportManager.java]: Could this be made private and final with a getter method if external access is required? Non-encapsulated fields are causing us issues on other changes to this code-base, so I'd prefer we don't introduce more.

**Pull request merged**